SquashFS iteration fails with "unexpected EOF"

**What happened**:

When iterating through a SquashFS layer within particular SIF images, an error is encountered. The image described in https://github.com/anchore/syft/issues/1150 is one example of this.

**What you expected to happen**:

Expected SquashFS iteration to succeed with well-formed SIF images.

**How to reproduce it (as minimally and precisely as possible)**:

Install [SingularityCE](https://github.com/sylabs/singularity) and build a SIF:

```sh
singularity build OneAPI.sif docker://intel/oneapi-hpckit:latest
```

Process with stereoscope:

```sh
$ go run examples/basic.go singularity:OneAPI.sif 
DEBU[0000] image: source=Singularity location=OneAPI.sif 
DEBU[0015] image metadata: digest=sha256:2f58e7c87a19782f3f459bd8db1393fb37740a38e78957228200a753d31de312 mediaType=application/vnd.sylabs.sif.layer.v1.sif tags=[] 
DEBU[0015] layer metadata: index=0 digest=sha256:11cc80431c3b72354f0a2c551508592b874579be91ab23a75343037f6a44f93e mediaType=application/vnd.sylabs.sif.layer.v1.squashfs 
panic: could not read image: failed to walk layer="sha256:11cc80431c3b72354f0a2c551508592b874579be91ab23a75343037f6a44f93e": open opt/intel/oneapi/compiler/2022.1.0/linux/lib/oclfpga/host/linux64/bin/perl/lib/5.30.3/pod/perlpod.pod: unexpected EOF

goroutine 1 [running]:
main.main()
        /home/adam/src/stereoscope/examples/basic.go:32 +0x73b
exit status 2
```

**Anything else we need to know?**:

The error occurs with or without the patch from #139 applied, so I believe this issue is distinct from #138.

**Environment**:

```sh
$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

```sh
$ singularity --version
singularity-ce version 3.10.2
```

--- a/go.mod
+++ b/go.mod
@@ -26,8 +26,8 @@ require (
 	github.com/sirupsen/logrus v1.8.1
 	github.com/spf13/afero v1.6.0
 	github.com/stretchr/testify v1.7.0
-	github.com/sylabs/sif/v2 v2.7.0
-	github.com/sylabs/squashfs v0.5.5-0.20220803150326-9393a0b4cef5
+	github.com/sylabs/sif/v2 v2.7.2
+	github.com/sylabs/squashfs v0.6.1
 	github.com/wagoodman/go-partybus v0.0.0-20200526224238-eb215533f07d
 	github.com/wagoodman/go-progress v0.0.0-20200621122631-1a2120f0695a
 	golang.org/x/crypto v0.0.0-20210921155107-089bfa567519

--- a/pkg/file/squashfs_walk.go
+++ b/pkg/file/squashfs_walk.go
@@ -16,12 +16,12 @@ import (
 // returns the special value fs.SkipDir, WalkSquashFS skips the current directory (path if
 // d.IsDir() is true, otherwise path's parent directory). Otherwise, if the function returns a non-
 // nil error, WalkSquashFS stops entirely and returns that error.
-type SquashFSVisitor func(fsys *squashfs.FS, path string, d fs.DirEntry) error
+type SquashFSVisitor func(fsys fs.FS, path string, d fs.DirEntry) error
 
 // WalkSquashFS walks the file tree within the SquashFS filesystem read from r, calling fn for each
 // file or directory in the tree, including root.
 func WalkSquashFS(r io.ReaderAt, fn SquashFSVisitor) error {
-	fsys, err := squashfs.NewSquashfsReader(r)
+	fsys, err := squashfs.NewReader(r)
 	if err != nil {
 		return err
 	}
@@ -33,7 +33,7 @@ func WalkSquashFS(r io.ReaderAt, fn SquashFSVisitor) error {
 // fn for each file or directory in the tree, including root. Callers should use WalkSquashFS
 // where possible, as this function is considerably less efficient.
 func WalkSquashFSFromReader(r io.Reader, fn SquashFSVisitor) error {
-	fsys, err := squashfs.NewSquashfsReaderFromReader(r)
+	fsys, err := squashfs.NewReaderFromReader(r)
 	if err != nil {
 		return err
 	}
@@ -48,6 +48,6 @@ func walkDir(fsys fs.FS, fn SquashFSVisitor) fs.WalkDirFunc {
 			return err
 		}
 
-		return fn(&fsys.(*squashfs.Reader).FS, path, d)
+		return fn(fsys, path, d)
 	}
 }

--- a/pkg/image/layer.go
+++ b/pkg/image/layer.go
@@ -3,6 +3,7 @@ package image
 import (
 	"archive/tar"
 	"bytes"
+	"errors"
 	"fmt"
 	"io"
 	"io/fs"
@@ -235,21 +236,26 @@ func (l *Layer) indexer(monitor *progress.Manual) file.TarIndexVisitor {
 }
 
 func (l *Layer) squashfsVisitor(monitor *progress.Manual) file.SquashFSVisitor {
-	return func(fsys *squashfs.FS, path string, d fs.DirEntry) error {
-		f, err := fsys.Open(path)
+	return func(fsys fs.FS, path string, d fs.DirEntry) error {
+		ff, err := fsys.Open(path)
 		if err != nil {
 			return err
 		}
-		defer f.Close()
+		defer ff.Close()
 
-		metadata, err := file.NewMetadataFromSquashFSFile(path, f.(*squashfs.File))
+		f, ok := ff.(*squashfs.File)
+		if !ok {
+			return errors.New("unexpected file type")
+		}
+
+		metadata, err := file.NewMetadataFromSquashFSFile(path, f)
 		if err != nil {
 			return err
 		}
 
 		var fileReference *file.Reference
 
-		switch f := f.(*squashfs.File); {
+		switch {
 		case f.IsSymlink():
 			fileReference, err = l.Tree.AddSymLink(file.Path(metadata.Path), file.Path(metadata.Linkname))
 			if err != nil {

